File: TN-POS-SYS/TN-SYS-WEB/src/features/shell/data/models/m_tb_shell_mod.ts
// 🇻🇳 Model đại diện cho một Module được đăng ký trong hệ thống
// 🇺🇸 Model representing a registered module in the system
import type { M_Db_Ett } from '../../../../core/models';

export interface M_Tb_Shell_Mod extends M_Db_Ett {
  // 🇻🇳 ID định danh module (e.g., 'pos', 'inv')
  // 🇺🇸 Module identifier (e.g., 'pos', 'inv')
  c_mod_id: string;

  // 🇻🇳 Tiêu đề hiển thị của module
  // 🇺🇸 Display title of the module
  c_title: string;

  // 🇻🇳 Tên icon (Material/Shoelace)
  // 🇺🇸 Icon name (Material/Shoelace)
  c_icon: string;

  // 🇻🇳 Đường dẫn gốc (Base Route)
  // 🇺🇸 Base route path
  c_route: string;

  // 🇻🇳 Thứ tự hiển thị trên Sidebar
  // 🇺🇸 Display order on Sidebar
  c_order: number;
}

File: TN-POS-SYS/TN-SYS-API/src/internal/model/shell/m_tb_shell_mod.go
// 🇻🇳 Shell Module Model
// 🇺🇸 Shell Module Model
package shell

import "tn-pos-sys-api/internal/model"

// M_Tb_Shell_Mod Module registry model
type M_Tb_Shell_Mod struct {
	CModID    string  `json:"c_mod_id" gorm:"column:c_mod_id;primaryKey"`
	CTitle    string  `json:"c_title" gorm:"column:c_title;not null"`
	CIcon     string  `json:"c_icon" gorm:"column:c_icon"`
	CParentID *string `json:"c_parent_id" gorm:"column:c_parent_id"`
	CRoute    string  `json:"c_route" gorm:"column:c_route;not null"`
	COrder    int     `json:"c_order" gorm:"column:c_order;default:0"`
	model.M_Db_Ett
}

// TableName Tên bảng
func (M_Tb_Shell_Mod) TableName() string {
	return "shell.qtb_shell_mod"
}
